multi-region-load-tests/multi-region-tests.py

```diff
@@ -0,0 +1,332 @@
+                        "passFailMetrics": {
+                            "condition1": {
+                                "clientmetric": "error",
+                                "aggregate": "percentage",
+                                "condition": ">",
+                                "value": 5
+                            },
+                            "condition2": {
+                                "clientmetric": "response_time_ms",
+                                "aggregate": "avg",
+                                "condition": ">",
+                                "value": 1000,
+                                "requestName": "Podcast page"
+                            }
+                        }
+                    },
+                    "environmentVariables": {
+                        "webapp": "demo-podcastwebapp.azurewebsites.net"
+                    }
+                }
+            )
+            # print(result)
+            # uploading files to the test
+            for file_name in file_names:
+                resultPoller = adminClient.begin_upload_test_file(
+                    test_id, file_name, open(file_name, "rb"))
+            tests[resource['location']] = test_id
+        except HttpResponseError as e:
+            print('Service responded with error: {}'.format(e.response.json()))
+
+    return tests
+
+
+def run_tests(credential, resources, tests):
+    """
+    For each resource
+        Get the LoadTestRunClient
+        Run the test
+    
+    For each resource 
+        Poll for the test run completion 
+        Wait for results file to be available
+        Call the extract_and_update_results method to extract the results and add required columns
+        Upload results to Blob storage 
+
+    Parameters
+    ----------
+    credential : ClientSecretCredential,
+    resources: list
+    tests: dict
+
+    Returns
+    -------
+    dict
+
+    Raises
+    ------
+    HttpResponseError
+        If starting the test run  fails
+    TimeoutError
+        If waiting for the results file timed out with error
+    """
+    test_runs = {}
+    for resource in resources:
+
+        runClient = LoadTestRunClient(
+            credential=credential, endpoint=resource['data_plane_uri'])
+        test_run_id = "test-run-" + str(int(time.time()))
+
+        try:
+            testRunPoller = runClient.begin_test_run(
+                test_run_id,
+                {
+                    "testId": tests[resource['location']],
+                    "displayName": "Load Test Run - "+resource['location']+" - "+str(int(time.time())),
+                }
+            )
+            test_runs[resource['location']] = {
+                "testRunPoller": testRunPoller, "test_run_id": test_run_id}
+
+        except HttpResponseError as e:
+            print("Failed with error: {}".format(e.response.json()))
+            exit(1)
+    print("Test runs")
+    print(test_runs)
+    time.sleep(500)
+
+    for resource in resources:

```

Comment:
The entire wait for result and upload part is done sequentially right? We can parallelize this